src/components/UploadAvatarModal/UploadAvatarModal.js

```diff
@@ -0,0 +1,247 @@
+        )
+
+        // Convert the contents of the canvas blob to an image
+        const reader = new FileReader()
+        canvas.toBlob(blob => {
+          reader.readAsDataURL(blob)
+          reader.onloadend = () => {
+            console.log(reader.result)
+            let arr = reader.result.split(','),
+            mime = arr[0].match(/:(.*?);/)[1],
+            bstr = atob(arr[1]),
+            n = bstr.length,
+            u8arr = new Uint8Array(n)
+
+            while(n--){
+              u8arr[n] = bstr.charCodeAt(n)
+            }
+            console.log(u8arr)
+            let croppedImage = new File([u8arr], 'avatar.png', {type:mime})
+
+            console.log(croppedImage.size)
+            if (croppedImage.size > 25*1024*1024) {
+              setResult({
+                error: { status: 'toobig' },
+                success: false,
+                submitted: false,
+              })
+            } else {
+              submit(mime, croppedImage)
+            }
+/*            // This is only here for debug
+            console.log("Loading preview")
+            const reader2 = new FileReader()
+            reader2.addEventListener('load', () => setPreviewImg(reader.result))
+            reader2.readAsDataURL(croppedImage)
+*/
+          }
+        })
+      })
+      image.src = upImg
+
+    } catch (e) {
+      console.error(e)
+      setResult({
+        error: { status: 'internal' },
+        success: false,
+        submitted: false,
+      })
+    }
+  }
+
+  const submit = async (mime, img) => {
+    setSubmitting(true)
+    const response = await dispatch(updateAvatar(userId, mime, img))
+    const error = getResponseError(response)
+
+    setResult({
+      error,
+      success: !error,
+      submitted: true,
+    })
+
+    if (!error) {
+      setTimeout(() => {
+        if (isOpen) {
+          onClose()
+        }
+      }, SUBMIT_AUTO_CLOSE_DELAY_TIME)
+    } else {
+      setSubmitting(false)
+    }
+  }
+
+  const userId = useSelector(selectCurrentUserId)
+
+  return (
+    <ModalContent className="dialog no-pad">
+      <UploadAvatarMessageBox result={result} />
+      <div>
+        <label>Select Image: </label>
+        <input type="file" accept="image/*" onChange={onSelectFile} />
+      </div>
+      <div class="zoomandcrop">
+        <div class='zoomsliderbox'>
+          <div class='zoomslidericon'>
+            <FontAwesomeIcon icon="search-plus" />
+          </div>
+          <div class='slidercontrol'>
+            <Slider
+              value={zoom}
+              min={1}
+              max={3}
+              step={0.1}
+              vertical={true}
+              aria-labelledby="Zoom"
+              onChange={(zoom) => { setZoom(zoom) }}
+            />
+          </div>
+          <div class='zoomslidericon'>
+            <FontAwesomeIcon icon="search-minus" />
+          </div>
+        </div>
+        <div class="rotateandcrop">
+          <div className="crop-container">
```

Comment:
One thing I noticed from your preview is that the copper is at an odd aspect ratio, with black bars on either side. was that a side effect of the page size or is it always like that? If it's always like that, we should probably see if it's possible to ensure it's always squared up unless page height/width restrictions require us to shrink it.